ui-v2/src/components/flows/cells.tsx

```diff
@@ -75,13 +80,80 @@ export const FlowDeploymentCount = ({ row }: { row: { original: Flow } }) => {
 	return data;
 };
 
+const DeleteMenuItem = ({
+	id,
+	name,
+	onClose,
+}: {
+	id: string;
+	name: string;
+	onClose?: () => void;
+}) => {
+	const [isOpen, setIsOpen] = useState(false);
+	const { toast } = useToast();
+
+	const onOpenChange = useCallback(
+		(isOpen: boolean) => {
+			if (!isOpen && onClose) {
+				onClose();
+			}
+			setIsOpen(isOpen);
+		},
+		[setIsOpen, onClose],
+	);
+
+	const onSelect = useCallback(
+		(event: Event) => {
+			event.preventDefault();
+			onOpenChange(true);
+		},
+		[onOpenChange],
+	);
+
+	const mutationOptions: MutationOptions = useMemo(
+		() => ({
+			onSuccess: () => {
+				toast({ title: "Flow Deleted" });
+				onOpenChange(false);
+			},
+			onError: (error) => {
+				const title = error.message || "Unknown error while deleting flow.";
+				toast({ title });
+			},
+		}),
+		[toast, onOpenChange],
+	);
+
+	const { mutate: deleteFlow, isPending } = useMutation(
+		deleteFlowMutation(id, mutationOptions),
+	);
```

Comment:
I noticed a [FlowQuery](https://github.com/PrefectHQ/prefect/blob/a8ac0e5acd416e3e36d890fe21f3ea930a8fcb52/ui-v2/src/components/flows/queries.tsx#L229) class that has a getter defined for this method.

Seems it is mostly intended to be use by the flows detail page, so went ahead and imported the method directly here as opposed to using the `FlowQuery`. This is because as more functionality is added to these cells / dropdown menu, probably not the best idea to new up a FlowQuery in every component that is taking an action on the flow.

However, I could see a world where the data coming into these cells is transformed a bit more, possibly leveraging the `FlowQuery` class. This would likely happen somewhere further upstream, either in the [route loader](https://github.com/PrefectHQ/prefect/blob/a8ac0e5acd416e3e36d890fe21f3ea930a8fcb52/ui-v2/src/routes/flows/index.tsx#L65-L66) or possibly the [flows data-table component](https://github.com/PrefectHQ/prefect/blob/a8ac0e5acd416e3e36d890fe21f3ea930a8fcb52/ui-v2/src/components/flows/data-table.tsx#L158)